src/pages/api/v1/youfocus/index.ts

```diff
@@ -0,0 +1,103 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { connectDB } from '@/middlewares';
+import { apiStatusCodes } from '@/constant';
+import { sendAPIResponse } from '@/utils';
+import {
+  extractPlaylistId,
+  fetchPlaylistData
+}
+  from '@/utils';
+import { checkPlaylistExistsByPlaylistId, addPlaylistToDB } from '@/database';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  await connectDB();
+
+  switch (req.method) {
+    case 'POST':
+      return handleAddPlaylist(req, res);
+    default:
+      return res.status(apiStatusCodes.BAD_REQUEST).json({
+        success: false,
+        message: `Method ${req.method} not allowed`,
+      });
+  }
+};
+
+const handleAddPlaylist = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { playlistUrl }: { playlistUrl: string } = req.body;
+
+  if (!playlistUrl) {
+    return res.status(apiStatusCodes.BAD_REQUEST).json({
+      success: false,
+      message: 'Playlist URL is required',
+    });
+  }
+
+  const playlistId = extractPlaylistId(playlistUrl);
+
+  if (!playlistId) {
+    return res.status(apiStatusCodes.BAD_REQUEST).json({
+      success: false,
+      message: 'Invalid playlist URL',
+    });
+  }
```

Comment:
why don't you simply check for playlist id. No need to check for id and URL both